test/test_linalg.py

```diff
@@ -8336,6 +8337,112 @@ def test_tensordot(self, device):
         an = torch.from_numpy(np.tensordot(np.zeros((), dtype=np.float32), np.zeros((), dtype=np.float32), 0))
         self.assertEqual(a, an)
 
+    @skipCUDAIfNoCusolver
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @skipCUDAIfRocm
+    @dtypes(*floating_and_complex_types())
+    def test_ldl_factor(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+        from scipy.linalg import ldl as scipy_ldl
+
+        def run_test(shape, batch, hermitian, upper):
+            A = random_hermitian_pd_matrix(shape, *batch, dtype=dtype, device=device)
+            actual_factors, actual_pivots, info = torch.linalg.ldl_factor_ex(A, hermitian=hermitian, upper=upper)
+            tri = torch.triu if upper else torch.tril
+            diag = 1 if upper else -1
+            actual_L = tri(actual_factors, diagonal=diag)
```

Comment:
A bit tricky to read. Perhaps a ternary operator is a better option here.

Reply:
`if ... else ...` is already used here to define `tri` and `diag`. Do you want this piece of code to be modified to
```py
actual_L = torch.triu(actual_factors, diagonal=diag) if upper else torch.tril(actual_factors, diagonal=diag)
```

Reply:
or a fully fledged  if-else if the ternary operator is too cluttered ternary operator, whatever you see fit. Either
`actual_L = torch.triu(actual_factors, diagonal=1) if upper else torch.tril(actual_factors, diagonal=-1)`
or 
```python
if upper:
  actual_L = torch.triu(actual_factors, diagonal=1)
else:
  actual_L = torch.tril(actual_factors, diagonal=-1)
```
Probably the if-else is more readable?